src/widget/slider.rs

```diff
@@ -200,9 +200,9 @@ impl<T: SliderType, D: Directional> Slider<T, D> {
             a = b - a;
         }
         let value = a + self.range.0;
-        let value = if !(value >= self.range.0) {
+        let value = if value < self.range.0 {
             self.range.0
-        } else if !(value <= self.range.1) {
+        } else if value > self.range.1 {
```

Comment:
These are not equivalent (consider NaN).

Reply:
Your solution here is not what [the lint](https://rust-lang.github.io/rust-clippy/master/index.html#neg_cmp_op_on_partial_ord) suggests — but that would be too verbose (impairing readability) here. I'm not entirely happy with just squelching the lint, but I guess for now it will do.